src/index.mjs

```diff
@@ -68,8 +69,10 @@ export default function (options) {
     } else {
       // If not, find all links and use IntersectionObserver.
       Array.from(options.el.querySelectorAll('a'), link => {
-        observer.observe(link);
-        toPrefetch.add(link.href);
+        if (!options.filter || options.filter(link.href)) {
```

Comment:
alternatively we could default to `() => true` so this becomes `Array.from(blah).filter(filter).forEach(...)`